text/nodejs/0014-yarn-berry.md

```diff
@@ -0,0 +1,249 @@
+support it has steadily increased.
+
+
+## Implementation
+
+
+### API
+
+`yarn-berry-install`
+
+Provides: `yarn_packages` OR `node_modules` Requires: `node`, during `build`
+
+The buildpack does not require `yarn` since the Berry release itself is meant
+to be checked into version control under `.yarn/releases`. Node v16.10 and
+later include
+[Corepack](https://nodejs.org/dist/latest/docs/api/corepack.html), which is the
+preferred method of managing package manager installations across projects. 
+
+
+A new order group will be added to the Nodejs language family buildpack,
+resulting in the following:
+
+```toml
+...
+
+[[order]]
+
+  [[order.group]]
+    id = "paketo-buildpacks/ca-certificates"
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/watchexec"
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/node-engine"
+
+  [[order.group]]
+    id = "paketo-buildpacks/yarn-berry-install"
+
+  [[order.group]]
+    id = "paketo-buildpacks/node-module-bom"
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/node-run-script"
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/node-start"
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/yarn-start"
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/procfile"
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/environment-variables"
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/image-labels"
+    optional = true
+
+[[order]]
+
+  [[order.group]]
+    id = "paketo-buildpacks/ca-certificates"
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/watchexec"
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/node-engine"
+
+  [[order.group]]
+    id = "paketo-buildpacks/yarn-install"
+
+  [[order.group]]
+    id = "paketo-buildpacks/node-module-bom"
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/node-run-script"
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/node-start"
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/yarn-start"
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/procfile"
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/environment-variables"
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/image-labels"
+    optional = true
+
+...
+
+```
+
+The `yarn-berry-install` group is placed before the traditional `yarn-install`
+group since it detects on the presence of `.yarnrc.yml` and both app types
+will contain `yarn.lock` files.
+
+### Detection
+
+- `.yarnrc.yml` was created with the intention of making it easy to detect
+  whether a project uses Yarn Classic or Berry (Berry ignores `.yarnrc`
+  completely), and should be used for the detection of `yarn-berry-install`.
+
+### Build
```

Comment:
Will `Build` create a filesystem-based (i.e. syft-based) SBOM like `yarn-install` does today?

Reply:
Good question. Yes, syft [supports yarn berry](https://github.com/anchore/syft/pull/868). That should be called out here explicitly.